builder/amazon/common/step_source_ami_info.go

```diff
@@ -16,14 +19,55 @@ import (
 type StepSourceAMIInfo struct {
 	SourceAmi          string
 	EnhancedNetworking bool
+	AmiFilters         AmiFilterOptions
+}
+
+// Build a slice of AMI filter options from the filters provided.
+func buildAmiFilters(input map[*string]*string) []*ec2.Filter {
+	var filters []*ec2.Filter
+	for k, v := range input {
+		filters = append(filters, &ec2.Filter{
+			Name:   k,
+			Values: []*string{v},
+		})
+	}
+	return filters
+}
+
+type imageSort []*ec2.Image
+
+func (a imageSort) Len() int      { return len(a) }
+func (a imageSort) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a imageSort) Less(i, j int) bool {
+	itime, _ := time.Parse(time.RFC3339, *a[i].CreationDate)
```

Comment:
Yeah: This was direct from terraform, so we'll be consistent :)

https://github.com/hashicorp/terraform/blob/1b2f553d52c6fb25fd65d5b4b81ffd4a2bd264f4/builtin/providers/aws/data_source_aws_ami.go#L299-L314